### PR TITLE
[Android] Use the new overlay and Bionic module from Swift 6

### PR DESCRIPTION
**Pull request by finagolfin**
The new overlay was merged into Swift 6 in swiftlang/swift#74758.

This supercedes the earlier #2707 with more complete support.

@hyp and @bnbarham, please review.

### CODE REVIEW

File: Sources/SwiftLibraryPluginProvider/LibraryPluginProvider.swift [hunk 1]
```diff
@@ -137,7 +139,7 @@ private func _loadLibrary(_ path: String) throws -> UnsafeMutableRawPointer {
 #else
 private func _loadLibrary(_ path: String) throws -> UnsafeMutableRawPointer {
   guard let dlHandle = dlopen(path, RTLD_LAZY | RTLD_LOCAL) else {
-    throw LibraryPluginError(message: "loader error: \(String(cString: dlerror()))")
+    throw LibraryPluginError(message: "loader error: \(String(cString: dlerror()!))")
```
Comment (compnerd) [inline Sources/SwiftLibraryPluginProvider/LibraryPluginProvider.swift:142]: Does this work with glibc? I thought that it was annotated with non null in some cases (at least I ran into one with foundation)

**Review by compnerd — APPROVED** [review]

**Comment by compnerd** [comment]
@swift-ci please test

File: Sources/SwiftLibraryPluginProvider/LibraryPluginProvider.swift (hunk 1 above)
Comment (finagolfin) [reply Sources/SwiftLibraryPluginProvider/LibraryPluginProvider.swift:142]: Yep, worked fine in Fedora 40 x86_64, where I tested it before submitting, which uses a recent glibc 2.39-17 from earlier this year. Bionic simply added the `_Nullable` annotation to the output in the last year, so it _has_ to be unwrapped for Android here, but unless glibc somehow can guarantee this is `_Nonnull`, this force unwrap should work for both.

**Comment by finagolfin** [comment]
@ahoppen, ready for merge.

**Comment by finagolfin** [comment]
Thanks for the quick review, @compnerd.

**Pull request merged**